cub/cub/block/block_merge_sort.cuh

```diff
@@ -91,30 +97,73 @@ _CCCL_DEVICE _CCCL_FORCEINLINE void SerialMerge(
   CompareOp compare_op,
   KeyT oob_default)
 {
+  _CCCL_ASSERT(keys1_count >= 0, "");
+  _CCCL_ASSERT(keys2_count >= 0, "");
+  if constexpr (IS_FULL_TILE)
+  {
+    _CCCL_ASSERT(keys1_count + keys2_count >= ITEMS_PER_THREAD, "");
+  }
+
+  // TODO(bgruber): this assertion fails. Is this a problem?
+  // _CCCL_ASSERT(keys1_count + keys2_count > 0, "");
```

Comment:
This would only be able to pass if either `!IS_FULL_TILE` or `ITEMS_PER_THREAD == 0`